docs/configuration/http_filters/router_filter.rst

```diff
@@ -120,6 +120,29 @@ Note that retry policies can also be applied at the :ref:`route level
 
 By default, Envoy will *not* perform retries unless you've configured them per above.
 
+x-envoy-grpc-retry-on
+^^^^^^^^^^^^^^^^^^^^^
+Setting this header on egress requests will cause Envoy to attempt to retry failed requests (number of
+retries defaults to 1, and is configured as x-envoy-grpc-retry is, above).  gRPC retries are currently
+only supported for gRPC status codes in response headers.  gRPC status codes in trailers will not trigger
+retry logic.One or more policies can be specified  using a ',' delimited list. The supported policies are:
+
+cancelled
```

Comment:
Cancelled are returned typically when the call are cancelled by the caller, not sure a proxy should retry. 

http://www.grpc.io/docs/guides/concepts.html#cancelling-rpcs